src/sql/src/plan/statement/ddl.rs

```diff
@@ -4213,7 +4214,16 @@ pub fn plan_grant_role(
         member_names,
     }: GrantRoleStatement<Aug>,
 ) -> Result<Plan, PlanError> {
-    let grantor_id = scx.catalog.active_role_id().clone();
+    // In PostgreSQL, the grantor must either be a role with ADMIN OPTION on the role being granted,
+    // or the bootstrap superuser. We do not have ADMIN OPTION implemented and 'mz_system' is our
+    // equivalent of the bootstrap superuser. Therefore the grantor is always 'mz_system'.
+    // For more details see:
+    // https://github.com/postgres/postgres/blob/064eb89e83ea0f59426c92906329f1e6c423dfa4/src/backend/commands/user.c#L2180-L2238
+    let grantor_id = scx
```

Comment:
Will this make the audit log useless for looking up which human typed grant or revoke? If so, we should delay this PR until ADMIN OPTION is implemented somehow. Can ADMIN OPTION be implied for superusers?

Reply:
> Will this make the audit log useless for looking up which human typed grant or revoke?

Pretty much yes.

> If so, we should delay this PR until ADMIN OPTION is implemented somehow.

Even when `ADMIN OPTION` is implemented, the grantor role does not always reflect the role that executed the `GRANT` statement. It reflects the role chosen as the best grantor.

> Can ADMIN OPTION be implied for superusers.

Yes in the sense that superusers can always execute `GRANT`. No in the sense that the superuser will not be chosen as the grantor unless they have `ADMIN OPTION` explicitly.

Reply:
Maybe the audit log entry should contain an additional field of something like "session role membership".

Reply:
I updated the audit log to also contain the role of the session that execute the command.

If you're curious the pg mailing thread that discusses this change can be found here: https://www.postgresql.org/message-id/flat/1102925.1654199458%40sss.pgh.pa.us#505bfe4947d7a86ee07d5fdad85f34d1

It's very long and mostly irrelevant to this change (I haven't finished reading the whole thing yet). Scroll down to where the email subject changes from "CREATEROLE and role ownership hierarchies" to "pg_auth_members.grantor is bunk" for the relevant emails.